pyproject.toml

```diff
@@ -1,5 +1,77 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+    "setuptools>=63",
+]
+
+[project]
+name = "arraycontext"
+version = "2021.1"
+description = "Choose your favorite numpy-workalike"
+readme = "README.rst"
+license = { text = "MIT" }
+authors = [
+    { name = "Andreas Kloeckner", email = "inform@tiker.net" },
+]
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Other Audience",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Utilities",
+]
+dependencies = [
+    "immutabledict>=4.1",
+    "numpy",
+    "pytools>=2024.1.3",
+]
+
+[project.optional-dependencies]
+jax = [
+    "jax>=0.4",
+]
+pyopencl = [
+    "islpy>=2024.1",
+    "loopy>=2024.1",
+    "pyopencl>=2024.1",
+]
+pytato = [
+    "pytato>=2021.1",
+]
```

Comment:
I added all of the various array context implementations as optional dependencies here. Let me know if you prefer them somewhere else!

Reply:
Potentially "fixes" #247 if left like this :grin:

Reply:
Putting those there seems OK. As for #247, there's still `call_loopy`, which is a bit questionable.

Reply:
Yeah, the code itself is still quite dependent on loopy, at least morally :(